### PR TITLE
tests: fix for tests test-*-cgroup

### PR DESCRIPTION
**Pull request by sergiocazzolato**
This change makes more simple the snaps used and changes the char device
used which was causing issues sporadically.

The errror in travis:
google:arch-linux-64 .../tests/main/security-device-cgroups-devmode#
test-devmode-cgroup.read-kmsg
head: error reading '/dev/kmsg': Broken pipe

### CODE REVIEW

File: tests/lib/snaps/test-devmode-cgroup/bin/read-dev [hunk 1]
```diff
@@ -0,0 +1,4 @@
+#!/bin/sh
+
+device=$1
+dd if="/dev/$device" of=/dev/null count=10
```
Comment (zyga) [inline tests/lib/snaps/test-devmode-cgroup/bin/read-dev]: Nitpick: missing newline

File: tests/lib/snaps/test-devmode-cgroup/bin/read-dev [hunk 2]
```diff
@@ -0,0 +1,4 @@
+#!/bin/sh
```
Comment (zyga) [inline tests/lib/snaps/test-devmode-cgroup/bin/read-dev]: Can you please make this 
```suggestion
#!/bin/sh -eu
```

File: tests/lib/snaps/test-devmode-cgroup/bin/read-dev [hunk 3]
```diff
@@ -0,0 +1,4 @@
...
+
+device=$1
```
Comment (zyga) [inline tests/lib/snaps/test-devmode-cgroup/bin/read-dev]:
```suggestion
device="$1"
```

File: tests/lib/snaps/test-strict-cgroup/bin/read-dev [hunk 4]
```diff
@@ -0,0 +1,4 @@
+#!/bin/sh
```
Comment (zyga) [inline tests/lib/snaps/test-strict-cgroup/bin/read-dev]: Can you please make this

```suggestion
#!/bin/sh -eu
```

File: tests/lib/snaps/test-strict-cgroup/bin/read-dev [hunk 5]
```diff
@@ -0,0 +1,4 @@
...
+
+device=$1
```
Comment (zyga) [inline tests/lib/snaps/test-strict-cgroup/bin/read-dev]:
```suggestion
device="$1"
```

File: tests/lib/snaps/test-strict-cgroup/bin/read-dev [hunk 6]
```diff
@@ -0,0 +1,4 @@
...
+dd if="/dev/$device" of=/dev/null count=10
```
Comment (zyga) [inline tests/lib/snaps/test-strict-cgroup/bin/read-dev]: Nitpick: please add a trailing newline.

File: tests/lib/snaps/test-devmode-cgroup/bin/read-dev (hunk 1 above)
Comment (sergiocazzolato) [reply tests/lib/snaps/test-devmode-cgroup/bin/read-dev]: done

File: tests/lib/snaps/test-devmode-cgroup/bin/read-dev (hunk 2 above)
Comment (sergiocazzolato) [reply tests/lib/snaps/test-devmode-cgroup/bin/read-dev]: done

File: tests/lib/snaps/test-devmode-cgroup/bin/read-dev (hunk 3 above)
Comment (sergiocazzolato) [reply tests/lib/snaps/test-devmode-cgroup/bin/read-dev]: done

File: tests/lib/snaps/test-strict-cgroup/bin/read-dev (hunk 4 above)
Comment (sergiocazzolato) [reply tests/lib/snaps/test-strict-cgroup/bin/read-dev]: done

File: tests/lib/snaps/test-strict-cgroup/bin/read-dev (hunk 6 above)
Comment (sergiocazzolato) [reply tests/lib/snaps/test-strict-cgroup/bin/read-dev]: done

File: tests/lib/snaps/test-strict-cgroup/bin/read-dev (hunk 5 above)
Comment (sergiocazzolato) [reply tests/lib/snaps/test-strict-cgroup/bin/read-dev]: done

**Comment by sergiocazzolato** [comment]
@zyga thanks for reviewing this, changes already implemented.

File: tests/lib/snaps/test-devmode-cgroup/bin/read-dev [hunk 7]
```diff
@@ -0,0 +1,4 @@
+#!/bin/sh -eu
+
+device="$1"
```
Comment (bboozzoo) [inline tests/lib/snaps/test-devmode-cgroup/bin/read-dev]: I'm not sure you need separate `device=..` now.

File: tests/lib/snaps/test-devmode-cgroup/bin/read-dev [hunk 8]
```diff
@@ -0,0 +1,4 @@
...
+dd if="/dev/$device" of=/dev/null count=10
```
Comment (bboozzoo) [inline tests/lib/snaps/test-devmode-cgroup/bin/read-dev]: That's definitely better than blindly caling `cat` on framebuffer :+1:

**Comment by bboozzoo** [comment]
Wondeing about that broken pipe error. We've taken care of this in `MATCH` already, so it's probably not coming from there. On top of this it's just `head -1`, not sure why that would fail. Maybe it's this: https://elixir.bootlin.com/linux/latest/source/kernel/printk/printk.c#L859 @zyga what do you think?

**Comment by zyga** [comment]
@bboozzoo not sure about the pipe, I think the patch makes a nice improvement even if it doesn't address the issue fully yet.

**Pull request merged**